android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt

```diff
@@ -475,6 +427,26 @@ class NetworkIntentService : IntentService("NetworkIntentService") {
         }
     }
 
+    private fun handleEnvoySubmit(
+        url: String,
+        captive_portal_url: String,
+        hysteriaCert: String?
+    ) {
+
+        // nothing to parse at this time, leave url in string format
+
+        httpsUrls.add(url)
+
+        // add a slight delay to give the direct connection a chance
+        Log.d(TAG, "submit url after a short delay for testing direct connection")
+        ioScope.launch() {
+            Log.d(TAG, "start envoy delay")
+            delay(5000L) // wait 5 seconds
```

Comment:
we should see if we could move to an event based model so we can sync these without putting waits in

Reply:
that's a fair point, i do wonder if we could ask os somehow, but probably not.  will resolve for now, should consider doing the dir con test in the near future though as cutting out a few seconds would be nice